### DISCUSSION TITLE
Welcome to Fancyapps UI Discussions!

### DISCUSSION THREAD

**Discussion by fancyapps** [fancyapps/ui#1] [Announcements]
## 👋 Welcome!

  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by P4KM4N** [comment]
First thank you for this amazing plugin! And congrats for this new awesome version!

I'm trying to add Fancyapps UI to a a WordPress theme I'm developing with Roots/Sage 10, who using Laravel Mix (webpack).

- I installed it with Yarn (I can see the folder in node_module).
- I setup the import in my app.js (main js file) like so:

```
import { Fancybox, Carousel, Panzoom } from "@fancyapps/ui";

$(document).ready(() => {

  Fancybox.bind("[data-fancybox]", {
    // Your options go here
  });
  console.log(Fancybox);

});
```

- In my console I get this message due to the console.log(Fancybox);
`class T extends r{constructor(t,i={}){L=function(){let t=!1;return document.createElement("div").focus({get preventScroll(){return t=!0,!1}}),t}(),super(function(t,i){const s=e(!0,{},t[i.startIndex]||{});r…`
But if I try to write Fancybox in the console I get:
```
VM767:1 Uncaught ReferenceError: Fancybox is not defined
    at <anonymous>:1:1
```
I'm clueless how to make it work, I'm trying for help here, sorry if it's not relevant, I will also ask for help on Roots forums, if I get answer I will post it here.

Thanks.

**Comment by fancyapps** [comment]
Hi,

That is actually how it should be. If you wish to make it accessible from anywhere (including console), simply make it global:

```js
window.Fancybox = Fancybox;
```

FYI, there is no need to use `$(document).ready(() => {})`, because Fancybox uses event delegation and it will work for all elements that match the selector, now or in the future.

**Reply by P4KM4N** [reply]
Wow! Thanks for that super fast answer Jānis! 
**_It work_** 🙌
I'm a real noob in JS, especially with vanilla.

**Comment by serg-by** [comment]
Hello!
How to add arrow control to carousel?

**Reply by fancyapps** [reply]
Hi, @serg-by ! Navigation arrows are created by the `Navigation` plugin and it is included by default. So you just have to have more than one page to display them. If you think you've found an use, please use https://github.com/fancyapps/ui/issues

**Reply by serg-by** [reply]
arrow control on the keyboard

**Reply by fancyapps** [reply]
Keyboard navigation is also enabled by default.

**Reply by serg-by** [reply]
strange but doesn't work. I cannot switch the slide using the arrows on the keyboard.

**Reply by Komarovski** [reply]
In my case navigation between slides with keyboard arrows also doesn't work. Here's how I'm implementing my carousel:
`new Carousel(document.getElementById("FancyCarouselFlexGallery"), {
	infinite:false,
	center:true,
	slidesPerPage:1,
	transition:"slide",
	Dots:false,
});`

**Reply by fancyapps** [reply]
@Komarovski Keyboard navigation for the Carousel is not provided "out of the box", you can easily add it yourself, example:

```js
const myCarousel = new Carousel(container, options);

document.addEventListener("keydown", (e) => {
  if (e.key === "ArrowLeft") {
    myCarousel.slidePrev();
  } else if (e.key === "ArrowRight") {
    myCarousel.slideNext();
  }
})
```
https://jsfiddle.net/g1h204xb/

**Reply by Komarovski** [reply]
@fancyapps Thank you very much! 👍

**Comment by laugepetersen** [comment]
How do we overwrite the default options for html instances? I can't make it work with the documentation.
 
I simply wanna add "MY_CUSTOM_CLASS" class to the "tpl" in the html5video object.

Playground Demo: https://fancyapps.com/playground/DL

**Reply by fancyapps** [reply]
Options for any plugin are included in the option named after the plugin, example: 

```JS
Fancybox.bind("[data-fancybox]", {
  Html : {
    // Html plugin optionss
  },
  Image : {
    // Image plugin options
 }
});
```

**Reply by laugepetersen** [reply]
Thanks, didn't get that anywhere in docs :)

Another one is how the ScrollLock plugin is activated then?

`ScrollLock: true` ?

**Reply by fancyapps** [reply]
ScrollLock is enabled by default, to disable, use `ScrollLock: false` (it does not work perfectly on iOS, sadly, a trillion dollar company does not care much about their software).

**Reply by laugepetersen** [reply]
Okay, that’s why it didn’t work.
Fixed with css 👍🏻

**Comment by laugepetersen** [comment]
Fancybox with Videojs minor bugs.
I've created it before where the fancybox v3 opens a video.js video.
But in this update v4 all the range/sliders input fields on the video such as volume bar and progress bar doesn't work within the fancybox. Any information on that?

**Reply by fancyapps** [reply]
Any demo?

**Reply by laugepetersen** [reply]
Yes, just quick sample of same settings as mine.

Fancybox Beta V4 with Video.js 7.15.

Demo: https://codepen.io/laugepetersen/pen/WNOeYqO?editors=1000

**Reply by fancyapps** [reply]
Add this snippet to disable touch actions:

```JS
"Carousel.Panzoom.touchStart": (fancybox, carousel, panzoom, event) => {
  if (fancybox.getSlide().$content.contains(event.target)) {
    return false;
  }
}
```

See this demo - https://fancyapps.com/playground/FY

btw, play.js is poorly designed, Plyr does not need this, because it uses semantic HTML.

**Reply by laugepetersen** [reply]
Thanks, worked out.

Do you mean video.js (play.js you wrote).

**Reply by fancyapps** [reply]
Yes, sorry, it is `video.js`

**Comment by nickelswitte** [comment]
Hi,

how do I set the maxScale of my FancyBox to another value? Do I have to do it via PanZoom or via Fancybox?
I find no example in the docs and unfortunately I am not experienced enough to deduce the answer myself.

Thanks!

**Reply by fancyapps** [reply]
Hi, What exactly do you want to archive? Do you want to increase the size of the image proportionally or do you want to choose custom sizes?

**Reply by nickelswitte** [reply]
When I open an image with fancybox and then click or touch once, the image is being zoomed in too much, and you completely lose the overview. I'd like to reduce the zoom level on the click / touch.

Thanks for the awesome library!

**Reply by fancyapps** [reply]
You can use `Panzoom.ratio` option. Almost any option can be passed as method and it will be executed when needed, therefore you can return value based on your custom logic. See this demo - https://fancyapps.com/playground/1dW - here, image size is limited to max 1.5x of the viewport. Or use `maxScale` like this - https://fancyapps.com/playground/1dY

**Reply by nickelswitte** [reply]
Hello,

thanks for the super fast response and the great reply. It works for me, when using this code snippet. There is still a thing I'd wonder if it is possible and that would be the absolute perfect solution:

Is it possible to configure fancybox like so, that the zoom on `click` or `touch` zooms in to a specified factor (so that you dont lose the overview of the image), but it is **still possible** to zoom im further after that using the mouse wheel or the gesture? 

With your currently proposed solution, I am zooming in to a much better factor, but can't go any further.

**Reply by fancyapps** [reply]
Yes, it is possible. Disable default click action and create your own that zooms in to the level you choose. It is very simple, see this demo - https://fancyapps.com/playground/1e8

**Comment by minhz** [comment]
i already include script to panzoom_plugins_toolbar.lean.js

but i cannot show toolbar panzoom ?
please help how do i show Transformation controls ?
like this:  https://fancyapps.com/fancybox/plugins/toolbar/


```
 <script type="text/javascript">
        Fancybox.bind('[data-fancybox]', {   //[data-fancybox="gallery"]
            Carousel: {
    Navigation: true,
  },
            Toolbar: { 
                display: [ 
                    { id: "counter", position: "center" }, 
                    "download", "zoomIn", "zoomOut", "iterateZoom", "zoom", "slideshow", "fullscreen", "thumbs", "close", 
                    ], 

            },

            Panzoom: {
                click: "toggleCover",
              Toolbar: {
                display: ["zoomIn", "zoomOut"],
              },

            }

        });
        </script>
```

**Reply by fancyapps** [reply]
@minhz If you are using v5, the API has completely changed. You can find examples on homepage.

**Comment by minhz** [comment]
is there anyway that i can add zoomIn zoomOut button ??

**Reply by fancyapps** [reply]
Is this a joke? There is a zoom in/out button by default. You can clearly see them on any demo - https://fancyapps.com/

**Reply by minhz** [reply]
hi no, sorry i'm newbie i mean seperate button Zoom IN and button Zoom out reduce/increase zoom

https://fancyapps.com/
Panzoom
Add panning and zooming gestures for any HTML content with ease. Present your content in a dynamic and interactive way with various transform controls.

**Reply by minhz** [reply]
also i tried mayn way but cannot add show these button

[
  "zoomIn",
  "zoomOut",
  "toggle1to1",
  "rotateCCW",
  "rotateCW",
  "flipX",
  "flipY",
  "reset",
  "toggleFullscreen",
];

**Reply by fancyapps** [reply]
It's as simple as copying/pasting code from the docs - 

```
Fancybox.bind('[data-fancybox="gallery"]', {
  Toolbar: {
    display: {
      left: ['infobar'],
      middle: [
        'zoomIn',
        'zoomOut',
        'toggle1to1',
        'rotateCCW',
        'rotateCW',
        'flipX',
        'flipY',
      ],
      right: ['slideshow', 'thumbs', 'close'],
    },
  },
});
```
https://stackblitz.com/edit/js-25nbpw?file=index.js

**Reply by minhz** [reply]
wow, many thanks for great works, let me try again. I tried before copy paste but now work, bad lucl for me

**Reply by fancyapps** [reply]
That's fine, I'll keep building and adding demos, and I'll add links to various code playgrounds to make it easier to understand the usage.

**Reply by minhz** [reply]
hi, i tried your code but it does not work (cannot click thumb image to see lightbox fancybox) in my wordpress

but only if i use this, it will works but not show those button flipX, flipY,...,...

```
<script src="https://cdn.jsdelivr.net/npm/@fancyapps/ui@4.0/dist/fancybox.umd.js"></script>
    <script src="https://cdn.jsdelivr.net/npm/@fancyapps/ui@5.0/dist/panzoom/panzoom.umd.js"></script>
<script type="text/javascript">
        Fancybox.bind('[data-fancybox]', {   //[data-fancybox="gallery"]
            items: {
download: {
    tpl: '<a title="{{DOWNLOAD}}" class="f-button" data-fancybox-download href="javasript:;"><svg><path d="M4 17v2a2 2 0 0 0 2 2h12a2 2 0 0 0 2-2v-2M7 11l5 5 5-5M12 4v12"/></svg></a>',
  },
            },
            Carousel: {
    Navigation: true,
  },    
Image: {
    Panzoom: {
      zoomFriction: 0.9,
      maxScale: function () {
        return 8;
      },
    },
  },
            Toolbar: { 
                display: [ 
                    { id: "counter", position: "center" }, 
                    { id: "prev", position: "center" },
                    { id: "next", position: "center" },
                    { id: "download", position: "center" }, 
                    "download", "flipX", "zoom", "slideshow", "fullscreen", "thumbs", "close", 
                    ], 
            },
        });
        </script>
```

**Reply by fancyapps** [reply]
The link to Fancybox5 JS file is like this - https://cdn.jsdelivr.net/npm/@fancyapps/ui@5.0/dist/fancybox/fancybox.umd.js

**Reply by fancyapps** [reply]
Also, your code is not compatible with v5! The syntax is different, see example I provided previously.

**Reply by minhz** [reply]
wow amazing, it works now, thanks for great works.

But now i'm trying to convert this from old version fancybox to new V5

```
    Image: {
    Panzoom: {
      zoomFriction: 0.9,
      maxScale: function () {
        return 8;
      },
    },
  },
```

**Reply by fancyapps** [reply]
Replace `Image: {` with `Images: {`

**Reply by minhz** [reply]
perfectly, thanks again for great works
